tonic/src/transport/service/discover.rs

```diff
@@ -9,7 +9,7 @@ use std::{
 };
 use tokio::sync::mpsc::Receiver;
 
-use tokio_stream::Stream;
+use futures_core::Stream;
```

Comment:
why this change?

Reply:
As tonic does not use tokio-stream except Stream which is the re-exporting of futures-core, we could use futures-core directly to reduce dependencies. This is detected by cargo-udeps.

Reply:
Can I ask the purpose of favoring `tokio-stream` with unused additional dependencies over `futures-core`?

Reply:
Ideally, we'd like to move off of the futures crates (stream is complicated since we expected it to land in std by now) since they are not as maintained as the tokio ones. Its a bit of a complicated history.

Reply:
Thanks for your answer. I am a bit confused, and would like to ask some more questions to better update this pull request. Even if we depend on `tokio-stream` instead of `futures-core`, as `Stream` we use is still provided by `futures-core`, we effectively depend on `futures-core`. Therefore I think that even if we use `tokio-stream` this time, we just add dependencies that we don't use and can't achieve our goal of not using `futures-core`. I would like to know a little more about this point.

Reply:
Eventually, `Stream` will go into `std` so then we will only depend on `tokio-stream` but for now we can't do that since progress has been slow there. That said, I would rather be positioned for that future than live with one extra dependency right now if that makes sense. In addition, we control tokio-stream so its better to depend on that then directly depend on another package if that makes sense. Happy to explain more over discord/email/what ever if you're interested 😄

Reply:
Thank you for your kind explanation!

> we control tokio-stream so its better to depend on that then directly depend on another package if that makes sense

Thanks to this point mainly, I think I reached to understand why we should depend on `tokio-steam` instead of `futures-core`. I will investigate whether it is possible to configure that we should explicitly depend on `tokio-stream`.

Reply:
@LucioFranco Should we replace the existing all `futures_core::Stream` with `tokio_stream::Stream` in this pull request, or should it be done in the another pull request?

Reply:
I found that the `cargo-udeps` checking will success by replacing all `futures_core::Stream` with `tokio_stream::Stream`! Therefore I think we should do it in this pull request.

Reply:
Yeah, we should convert everything from futures crates over. I think I originally used it but would like to remove it but I have not had time. Feel free to start on that!